cli/src/cmd/forge/script/build.rs

```diff
@@ -135,7 +135,10 @@ impl ScriptArgs {
 
                 // if it's the target contract, grab the info
                 if extra.no_target_name {
-                    if id.source == std::path::Path::new(&extra.target_fname) {
+                    if id.source ==
+                        dunce::canonicalize(&extra.target_fname)
+                            .expect("Could not canonicalize target path")
```

Comment:
looking at where this is originally created, looks like this is already `canonicalized`?